From 8089d78349dab554d6a036336e69614990ef6642 Mon Sep 17 00:00:00 2001
From: Nikita Pavlov <“niikitta@yahoo.com”>
Date: Fri, 22 Jul 2022 11:13:36 +0300
Subject: [PATCH] power operation changes

---
 .../ServerPowerOperations/ServerPowerOperations.vue       | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/views/Operations/ServerPowerOperations/ServerPowerOperations.vue b/src/views/Operations/ServerPowerOperations/ServerPowerOperations.vue
index 9e03083..973d48a 100644
--- a/src/views/Operations/ServerPowerOperations/ServerPowerOperations.vue
+++ b/src/views/Operations/ServerPowerOperations/ServerPowerOperations.vue
@@ -83,14 +83,14 @@
               <b-form-group
                 :label="$t('pageServerPowerOperations.rebootServer')"
               >
-                <b-form-radio
+                <!--<b-form-radio
                   v-model="form.rebootOption"
                   name="reboot-option"
                   data-test-id="serverPowerOperations-radio-rebootOrderly"
                   value="orderly"
                 >
                   {{ $t('pageServerPowerOperations.orderlyReboot') }}
-                </b-form-radio>
+                </b-form-radio>-->
                 <b-form-radio
                   v-model="form.rebootOption"
                   name="reboot-option"
@@ -113,14 +113,14 @@
               <b-form-group
                 :label="$t('pageServerPowerOperations.shutdownServer')"
               >
-                <b-form-radio
+                <!--<b-form-radio
                   v-model="form.shutdownOption"
                   name="shutdown-option"
                   data-test-id="serverPowerOperations-radio-shutdownOrderly"
                   value="orderly"
                 >
                   {{ $t('pageServerPowerOperations.orderlyShutdown') }}
-                </b-form-radio>
+                </b-form-radio>-->
                 <b-form-radio
                   v-model="form.shutdownOption"
                   name="shutdown-option"
-- 
2.30.2

