From 38244f76931c2bafb6fcc8b6b7398978aa742e8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E2=80=9CNikita?= <“niikitta@yahoo.com”>
Date: Mon, 14 Feb 2022 13:59:13 +0300
Subject: [PATCH] Static IP address for testing

---
 src/network_config.cpp | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/src/network_config.cpp b/src/network_config.cpp
index 94c8eae..79abbba 100644
--- a/src/network_config.cpp
+++ b/src/network_config.cpp
@@ -26,19 +26,7 @@ void writeDHCPDefault(const std::string& filename, const std::string& interface)
     // file is not present, or after the default file has been
     // manually removed.
     filestream << "[Match]\nName=" << interface <<
-                "\n[Network]\nDHCP=true\n"
-#ifdef LINK_LOCAL_AUTOCONFIGURATION
-                "LinkLocalAddressing=yes\n"
-#else
-                "LinkLocalAddressing=no\n"
-#endif
-#ifdef ENABLE_IPV6_ACCEPT_RA
-                "IPv6AcceptRA=true\n"
-#else
-                "IPv6AcceptRA=false\n"
-
-#endif
-                "[DHCP]\nClientIdentifier=mac\n";
+                "\n[Network]\nAddress=192.168.8.22/24\nGateway=192.168.8.1\n";
     filestream.close();
 }
 } // namespace bmc
-- 
2.30.2

