From 13a0a74a220313c1329a6c7bd840f7b8d63a2989 Mon Sep 17 00:00:00 2001
From: Nikita Pavlov <“niikitta@yahoo.com”>
Date: Wed, 29 Jun 2022 13:26:31 +0300
Subject: [PATCH] make property rw

---
 src/EntityManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/EntityManager.cpp b/src/EntityManager.cpp
index cc6aa12..b2cf45a 100644
--- a/src/EntityManager.cpp
+++ b/src/EntityManager.cpp
@@ -278,7 +278,7 @@ void populateInterfaceFromJson(
     std::shared_ptr<sdbusplus::asio::dbus_interface>& iface,
     nlohmann::json& dict, sdbusplus::asio::object_server& objServer,
     sdbusplus::asio::PropertyPermission permission =
-        sdbusplus::asio::PropertyPermission::readOnly)
+        sdbusplus::asio::PropertyPermission::readWrite)
 {
     for (auto& dictPair : dict.items())
     {
-- 
2.30.2

