diff --git a/control/service_files/json/phosphor-fan-control@.service.in b/control/service_files/json/phosphor-fan-control@.service.in
index 9b305cd..80a5ae6 100644
--- a/control/service_files/json/phosphor-fan-control@.service.in
+++ b/control/service_files/json/phosphor-fan-control@.service.in
@@ -1,5 +1,6 @@
 [Unit]
 Description=Phosphor Fan Control Daemon
+After=phosphor-fan-monitor@.service xyz.openbmc_project.Inventory.Manager.service
 
 [Service]
 Restart=on-failure
