From 05b1b67ac7b5388d07f108b655aba1e9e9de8b56 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E2=80=9CNikita?= <“niikitta@yahoo.com”>
Date: Mon, 14 Feb 2022 10:01:21 +0300
Subject: [PATCH] Change settings for use 64MiB image

---
 include/configs/ast-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/ast-common.h b/include/configs/ast-common.h
index eff6d2b529..2ae16bea96 100644
--- a/include/configs/ast-common.h
+++ b/include/configs/ast-common.h
@@ -108,7 +108,7 @@
 #define CONFIG_ENV_OFFSET		0x60000	/* environment starts here  */
 #define CONFIG_ENV_SIZE			0x20000	/* Total Size of Environment Sector */
 
-#define CONFIG_BOOTCOMMAND		"bootm 20080000"
+#define CONFIG_BOOTCOMMAND		"bootm 20100000"
 #define CONFIG_ENV_OVERWRITE
 
 #define ASPEED_ENV_SETTINGS \
-- 
2.30.2

